src/components/modals/SuggestModal.tsx

```diff
@@ -77,42 +75,21 @@ export function SuggestModal({ handleClose, isOpen }: SuggestModalProps) {
     }
   };
 
-  const handleSubmit = (e: FormEvent) => {
+  const handleSubmit = async (e: FormEvent) => {
     e.preventDefault();
 
     const isUrlValid = validateUrl(formValues.url);
     if (!isUrlValid) return;
 
-    createPR.mutate(
-      {
-        directoryPath: config.defaultDirectoryPath,
-        fileName: formValues.title,
-        url: formValues.url,
-        transcribedText: "",
-        prRepo: getPRRepo(),
-        needs: "transcript",
-      },
-      {
-        onError: (e) => {
-          const description =
-            e instanceof Error
-              ? e.message
-              : "An error occurred while submitting suggestion";
-          toast({
-            status: "error",
-            title: "Error submitting",
-            description,
-          });
-        },
-        onSuccess: () => {
-          toast({
-            status: "success",
-            title: "Suggestion submitted successfully",
-          });
-          resetAndCloseForm()
-        },
+    await suggestSource.mutateAsync({
```

Comment:
You will find the callback inside the `useGitHub` hook.
I added all the callbacks inside of the hooks for better separation of concerns. This `onSuccess` is the only one that "survived" because it uses context from the component.